src/Skoruba.MultiTenant.Identity/Validators/RoleRequiresTenantIdValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Skoruba.MultiTenant.Abstractions;
+using System.Threading.Tasks;
+
+namespace Skoruba.MultiTenant.Identity.Validators
+{
+    public class RoleRequiresTenantIdValidator<TRole> : IRoleValidator<TRole>
+        where TRole : class
+    {
+        private readonly ISkorubaTenantContext _skorubaTenantContext;
+
+        public RoleRequiresTenantIdValidator(ISkorubaTenantContext skorubaTenantContext)
+        {
+            _skorubaTenantContext = skorubaTenantContext;
+        }
+
+        public Task<IdentityResult> ValidateAsync(RoleManager<TRole> manager, TRole role)
+        {
+            if (_skorubaTenantContext.TenantResolutionRequired && !_skorubaTenantContext.TenantResolved)
+            {
+                throw MultiTenantException.MissingTenant;
```

Comment:
The reason i'm throwing an exception is because **it should not happen**.  If it does not resolve and is required then there is something wrong with the multi tenant implementation.  I worry that a graceful exit might hide a larger problem.

Reply:
Agreed.